sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosTracerTest.java

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos;
+
+import com.azure.core.util.Context;
+import com.azure.core.util.tracing.Tracer;
+import com.azure.cosmos.implementation.CosmosItemProperties;
+import com.azure.cosmos.implementation.LifeCycleUtils;
+import com.azure.cosmos.implementation.TestConfigurations;
+import com.azure.cosmos.implementation.TracerProvider;
+import com.azure.cosmos.implementation.directconnectivity.ReflectionUtils;
+import com.azure.cosmos.models.CosmosItemRequestOptions;
+import com.azure.cosmos.models.CosmosStoredProcedureProperties;
+import com.azure.cosmos.models.CosmosTriggerProperties;
+import com.azure.cosmos.models.CosmosUserDefinedFunctionProperties;
+import com.azure.cosmos.models.FeedOptions;
+import com.azure.cosmos.models.PartitionKey;
+import com.azure.cosmos.models.TriggerOperation;
+import com.azure.cosmos.models.TriggerType;
+import com.azure.cosmos.rx.TestSuiteBase;
+import org.mockito.Matchers;
+import org.mockito.Mockito;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.ServiceLoader;
+import java.util.UUID;
+
+public class CosmosTracerTest extends TestSuiteBase {
+    private static final String ITEM_ID = "tracerDoc";
+    CosmosAsyncClient client;
+    CosmosAsyncDatabase cosmosAsyncDatabase;
+    CosmosAsyncContainer cosmosAsyncContainer;
+
+    @BeforeClass(groups = {"emulator"}, timeOut = SETUP_TIMEOUT)
+    public void beforeClass() {
+        client = new CosmosClientBuilder()
+            .endpoint(TestConfigurations.HOST)
+            .key(TestConfigurations.MASTER_KEY)
+            .directMode(DirectConnectionConfig.getDefaultConfig())
+            .buildAsyncClient();
+        cosmosAsyncDatabase = getSharedCosmosDatabase(client);
+        cosmosAsyncContainer = getSharedMultiPartitionCosmosContainer(client);
+
+    }
+
+    @Test(groups = {"emulator"}, timeOut = TIMEOUT)
+    public void cosmosAsyncClient() {
+        TracerProvider tracer = Mockito.spy(new TracerProvider(ServiceLoader.load(Tracer.class)));
+        ReflectionUtils.setTracerProvider(client, tracer);
+
+        client.createDatabaseIfNotExists(cosmosAsyncDatabase.getId()).block();
+        Mockito.verify(tracer, Mockito.times(1)).startSpan(Matchers.anyString(), Matchers.anyString(),
```

Comment:
Rather than any string could we validate the span names and attributes being set on the span?

Reply:
Will address all testing format comment in next  tracer event pr as discussed offline